jupyterhub/orm.py

```diff
@@ -64,6 +64,8 @@ class Server(Base):
     """
     __tablename__ = 'servers'
     id = Column(Integer, primary_key=True)
+    
+    name = Column(Unicode(32), unique=True) # this is a unique reference to the server, probably a UUID4
```

Comment:
I think we can probably leave the constraint out of the database, and enforce it only at the code level. `Server.name` should probably be human-readable, and doesn't need to be unique. I think it's fine for several Servers to have no name, for instance.